cvat-sdk/cvat_sdk/core/uploading.py

```diff
@@ -335,7 +336,15 @@ def upload_files(
         if pbar is not None:
             pbar.start(total_size, desc="Uploading data")
 
-        self._tus_start_upload(url)
+        upload_info = None
+        if str(kwargs.get("sorting_method")).lower() == "predefined":
+            # Request file ordering, because we reorder files to send more efficiently
+            upload_info = {
+                "tus_file_order": [p.name for p in resources],
```

Comment:
Why is `tus_file_order` sent with the first request? It seems like the API would be simpler if it was sent with the last request, where all the other settings are. Do we need the file order before the upload is complete?

Reply:
The idea is that the first request distinguishes between recording the input order and a custom order. In principle, the required order can be sent later.

Reply:
IMO, it would be better if the order was sent in the last request instead of the first:

* The mental model for the user is simpler: all settings go in the final request. The client code becomes simpler as well. There's no `image_quality` duplication.
* The server code should become simpler as well. There's no need to save the custom file order to disk -> `_TUS_CUSTOM_ORDERING_FILE_NAME` and everything related to it can be removed. `upload_started` can be reduced to just `self._init_tus_input_ordering()`. Checks for whether `_TUS_INPUT_ORDERING_FILE_NAME` exists can be removed. The big comment in `DataSerializer` is no longer relevant.

The only downside I see is that you will end up needlessly saving the input order in the cases where a custom order is supplied, but like... you're already doing that in all cases when the sorting is _not_ predefined, so does it really matter if we do it in those cases as well? It shouldn't be cost-prohibitive to just record the order of input files.